app/controllers/submissions_controller.rb

```diff
@@ -0,0 +1,66 @@
+class SubmissionsController < ApplicationController
+  layout "users"
+  before_action :current_user
+
+  def create
+    @submission = Submission.new(submission_params)
+    if @submission.save
+      undo_link = view_context.link_to('UNDO', circle_submission_path(circle_id: @submission.circle_id, id: @submission.id), method: :delete).html_safe
+      flash[:success] = "Submission successfully created - #{undo_link}"
+    else
+      flash[:error] = ["Sorry, something went wrong. Please try again",
+                      @submission.errors.full_messages.to_sentence]
+    end
+    redirect_to circle_path(@submission.circle)
+  end
+
+  def edit_remix
+    @submission = Submission.find_by(circle_id: params[:id], original_id: params[:stem_id])
+    @tracks = @current_user.stems
+  end
+
+  def update
+    find_submission
+    if @submission.update_attributes(submission_params)
+      undo_link = view_context.link_to('UNDO', delete_remix_path(id: @submission.circle_id, stem_id: @submission.original.id), method: :delete).html_safe
```

Comment:
Line is too long. [153/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.